packages/misc/src/lib/misc.ts

```diff
@@ -410,6 +411,48 @@ export const checkType = ({
   return true;
 };
 
+/**
+ * Check if the given value complies with the given schema
+ * If not, throw `invalidParamType` error
+ *
+ * @param { any } value
+ * @param { JSONSchemaType<any> } schema
+ * @param { string } paramName
+ * @param { string } functionName
+ * @param { boolean } throwOnError
+ *
+ * @returns { Boolean } true/false
+ */
+export const checkSchema = (
+  value: any,
+  schema: JSONSchemaType<any>,
+  paramName: string,
+  functionName: string,
+  throwOnError = true
+) => {
+  const ajv = new Ajv();
```

Comment:
nit: this looks like a function that is called many times. Can we lazy load `ajv` such that we do not keep creating new instances each time? the implementation looks to be non mutative.

Reply:
Removed to a common place. Also made the schemas avoid compiling if Ajv has them already in the cache